### DISCUSSION TITLE
Derive man pages and manual from same source

### DISCUSSION THREAD

**Discussion by whitney-cumber** [neomutt/neomutt#4528] [Development]
Currently, the manual and the man pages have two different sources. This means there is **alot** of duplication both in information and verbatim text. It would be better to have a single source of truth and derive both the manual and the man page from that.
This would avoid mistakes like forgetting to port a modification from one version to the other (which I have encountered multiple times).

**Comment by flatcap** [comment]
That would be lovely, but nobody's come up with a plan of how to do it.
The current doc build looks like this: (see: https://neomutt.org/dev/build/make)

![build](https://neomutt.org/images/make-docs.svg)

(This is actually simpler than it used to be in Mutt!)
**Note**: This excludes the painful steps required to generate the Guide on the website.

Some of the docs are derived from the code and some bit of the docs can be omitted if not configured into NeoMutt.

**Comment by flatcap** [comment]
A fanciful plan I had once was Markdown + fake mailboxes.

All the docs would be in Markdown -- it's easily understood by beginners and has enough syntax for our needs.
Markdown can be used directly on the website.
Also each page would have an "Edit this page" button on it, encouraging users to contribute
(it would link to the page in the GitHub repo).

We could use `pandoc` to create man pages from the Markdown.

Rather than generate HTML docs for the user, the markdown could be viewed **in** NeoMutt.
We created a proof-of-concept in the [`help-backend`](https://github.com/neomutt/neomutt-old/tree/help-backend) branch.
It presents all the markdown as a fake "help" mailbox that you can read, search, etc.
Again, on every page you could `<edit-this-page>` and any user could potentially improve the docs.

This plan, though complicated, would separate the code and docs, but still allow the user to get help within NeoMutt.
